pkg/api/v1alpha1/provider_spec.go

```diff
@@ -35,6 +35,10 @@ type GCPProviderSpec struct {
 	// Enabling IP Forwarding.
 	CanIPForward bool `json:"canIpForward"`
 
+	// DualStack enables the creation of machines with both IPv4 and IPv6 support.
+	// This allows seamless integration into modern, dual-stack network environments.
+	DualStack bool `json:"dualStack"`
+
```

Comment:
MCM providers are kinda considered lower level API. So:

- I would consider adding them to the [NIC](https://github.com/gardener/machine-controller-manager-provider-gcp/blob/09ae5f9b18db08c0e76e8a340e338ed8148715d1/pkg/api/v1alpha1/provider_spec.go#L255). It either way seems like for GCP the "dual stack" is part of the NIC settings.
- I would expose them as is: meaning 2 different settings mirroring what the GCP client provides. provider-gcp can then configure it accordingly.